platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -623,6 +627,32 @@ private boolean isSignatureValid(@Nullable final Address address, @NonNull final
                 state.getHash().getBytes(), signature.getBytes(), address.getSigPublicKey());
     }
 
+    /**
+     * Check if a signature is valid. If a node has no weight or is missing a certificate, we consider the signature to
+     * be invalid.
+     *
+     * @param rosterEntry the roster entry of the signer, or null if there was no signing address
+     * @param signature   the signature to check
+     * @return true if the signature is valid, else false
+     */
+    private boolean isSignatureValid(@Nullable final RosterEntry rosterEntry, @NonNull final Signature signature) {
+        if (rosterEntry == null) {
```

Comment:
Line comments from the equivalent method above were not copied over. They will be lost when the address book version is cleaned up. I find them useful.